File: lib/models/node.dart
import 'package:logic_gates/models/gate.dart';

abstract class Node {
  final List<Node> next = [];
  final List<Node> previous = [];

  void addNext(Node node) {
    next.add(node);
    node.previous.add(this);
  }

  bool evaluate() {
    throw 'Missing implementation.';
  }
}

File: pkg/models/node.go
package models

type Node struct {
	Previous *Node
	Next     []*Node
}

func (n *Node) AddNext(next *Node) {
	n.Next = append(n.Next, next)
}

func (n *Node) Evaluate() bool {
	return false
}

func (n *Node) Forward() {}
